File: programmers/kakao/level2/n진수_게임.py
def convert_notation(n, k):
    s = '0123456789ABCDEF'
    q, r = divmod(n, k)

    return convert_notation(q, k) + s[r] if q else s[r]

def solution(n, t, m, p):
    answer = ""
    st = ""
    for i in range(t*m):
        st += convert_notation(i, n)
    st = st[:t*m]
    for i in range(p-1, len(st), m):
        answer += st[i]
    return answer


n, t, m, p = 2, 4, 2, 1
print(solution(n, t, m, p)) # "0111"

n, t, m, p = 16, 16, 2, 1
print(solution(n, t, m, p)) # "02468ACE11111111"

n, t, m, p = 16, 16, 2, 2
print(solution(n, t, m, p)) # "13579BDF01234567"

File: Java/programmers/kakao/level2/n진수게임.java
package CodeTest.Java.programmers.kakao.level2;

public class n진수게임 {
    public String solution(int n, int t, int m, int p) {
        StringBuilder answer = new StringBuilder();
        
        StringBuilder st = new StringBuilder();
        for (int i=0; i<t*m; i++) {
            st.append(convert_notation(i, n));
        }
        
        for (int i=p-1; i<t*m; i+=m) {
            answer.append(st.charAt(i));
        }
        
        return answer.toString();
    }
    
    private String convert_notation(int n, int k) {
        String[] s = "0123456789ABCDEF".split("");
        int q = n/k, r = n%k;
        
        return q != 0 ? convert_notation(q, k) + s[r] : s[r];
    }
}
